### PR TITLE
fix error when editing voltage level in line attach to vl creation

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/components/dialogs/network-modifications/line-attach-to-voltage-level/line-attach-to-voltage-level-dialog.js [hunk 1]
```diff
@@ -249,6 +256,18 @@ const LineAttachToVoltageLevelDialog = ({
...
+            fetchVoltageLevelsListInfos(studyUuid, currentNode?.id).then(
+                (values) => {
+                    setVoltageLevelOptions(
+                        values.sort((a, b) => a?.id?.localeCompare(b?.id))
+                    );
+                }
+            );
+        }
+    }, [studyUuid, currentNode?.id]);
+
```
Comment (ghelloumi) [inline src/components/dialogs/network-modifications/line-attach-to-voltage-level/line-attach-to-voltage-level-dialog.js:270]: same code bloc in `components/dialogs/network-modifications/line-split-with-voltage-level/line-split-with-voltage-level-dialog.js` 
please find a way to create only one function.

File: src/components/dialogs/network-modifications/line-attach-to-voltage-level/line-attach-to-voltage-level-form.js [hunk 2]
```diff
@@ -122,19 +100,28 @@ const LineAttachToVoltageLevelForm = ({
...
             studyUuid={studyUuid}
             currentNode={currentNode}
-            onVoltageLevelChangeCallback={onVoltageLevelChange}
         />
     );
 
+    const isVoltageLevelEdit =
+        voltageLevelToEdit &&
+        voltageLevelToEdit.equipmentId === voltageLevelIdWatch;
+
```
Comment (ghelloumi) [inline src/components/dialogs/network-modifications/line-attach-to-voltage-level/line-attach-to-voltage-level-form.js:124]:
```suggestion
    const isVoltageLevelEdit =
        voltageLevelToEdit &&
        voltageLevelToEdit.equipmentId === voltageLevelIdWatch;

```

const isVoltageLevelEdit = voltageLevelToEdit?.equipmentId === voltageLevelIdWatch;

File: src/components/dialogs/network-modifications/line-attach-to-voltage-level/line-attach-to-voltage-level-form.js [hunk 3]
```diff
@@ -192,7 +179,9 @@ const LineAttachToVoltageLevelForm = ({
                     currentNode={currentNode}
                     studyUuid={studyUuid}
                     onCreateVoltageLevel={onVoltageLevelCreationDo}
-                    editData={voltageLevelToEdit}
+                    editData={
```
Comment (ghelloumi) [inline src/components/dialogs/network-modifications/line-attach-to-voltage-level/line-attach-to-voltage-level-form.js:182]: type of 'undefined' is not an object
but in VoltageLevelCreationDialog we have editData: PropTypes.object,

File: src/components/dialogs/network-modifications/line-attach-to-voltage-level/line-attach-to-voltage-level-form.js [hunk 4]
```diff
@@ -192,7 +179,9 @@ const LineAttachToVoltageLevelForm = ({
...
+                        isVoltageLevelEdit ? voltageLevelToEdit : undefined
+                    }
```
Comment (ghelloumi) [inline src/components/dialogs/network-modifications/line-attach-to-voltage-level/line-attach-to-voltage-level-form.js:184]: type of 'undefined' is not an object
but in VoltageLevelCreationDialog we have editData: PropTypes.object,

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: small changes requested if you agree
tests: ok (both modifications)

File: src/components/dialogs/network-modifications/line-split-with-voltage-level/line-split-with-voltage-level-dialog.js [hunk 5]
```diff
@@ -211,6 +230,26 @@ const LineSplitWithVoltageLevelDialog = ({
                     preparedVoltageLevel.sectionCount,
                     preparedVoltageLevel.busbarCount
                 );
+                // we keep the old voltage level id, so it can be removed for from voltage level options
+                const oldVoltageLevelId = newVoltageLevel?.equipmentId;
+
+                const formattedVoltageLevel =
+                    getNewVoltageLevelData(preparedVoltageLevel);
+
+                // we add the new voltage level, (or replace it if it exists). And we remove the old id if it is different (in case we modify the id)
```
[the remark this answers is not in the document]
Comment (YenguiSeddik) [reply src/components/dialogs/network-modifications/line-split-with-voltage-level/line-split-with-voltage-level-dialog.js:240]: done

File: src/components/dialogs/network-modifications/line-split-with-voltage-level/line-split-with-voltage-level-form.js [hunk 6]
```diff
@@ -90,19 +70,27 @@ const LineSplitWithVoltageLevelForm = ({
         <TextInput name={LINE2_NAME} label={'Line2Name'} />
     );
 
+    const busbarSectionOptions = useMemo(() => {
+        if (voltageLevelIdWatch === voltageLevelToEdit?.equipmentId) {
```
[the remark this answers is not in the document]
Comment (YenguiSeddik) [reply src/components/dialogs/network-modifications/line-split-with-voltage-level/line-split-with-voltage-level-form.js]: done

File: src/components/dialogs/network-modifications/line-attach-to-voltage-level/line-attach-to-voltage-level-dialog.js [hunk 7]
```diff
@@ -287,6 +306,28 @@ const LineAttachToVoltageLevelDialog = ({
...
+                    formattedVoltageLevel.id === oldVoltageLevelId
+                        ? voltageLevelOptions.filter(
+                              (vl) => vl.id !== formattedVoltageLevel.id
+                          )
+                        : voltageLevelOptions.filter(
+                              (vl) =>
+                                  vl.id !== formattedVoltageLevel.id &&
+                                  vl.id !== oldVoltageLevelId
+                          );
+                newVoltageLevelOptions.push(formattedVoltageLevel);
```
[the remark this answers is not in the document]
Comment (YenguiSeddik) [reply src/components/dialogs/network-modifications/line-attach-to-voltage-level/line-attach-to-voltage-level-dialog.js]: done

**Review by ayolab — APPROVED** [review]

**Review by dbraquart — APPROVED** [review]
code: ok
tests: ok

**Pull request merged**